iac-common/src/test/java/org/sonar/iac/common/testing/AbstractMetricsTest.java

```diff
@@ -71,12 +71,16 @@ protected MetricsVisitor scan(String... codeLines) {
   }
 
   protected MetricsVisitor scan(String code) {
-    inputFile = new TestInputFileBuilder("moduleKey", new File(tempFolder, "file").getName())
+    inputFile = new TestInputFileBuilder("moduleKey", new File(tempFolder, testFileName()).getName())
       .setCharset(StandardCharsets.UTF_8)
       .setContents(code)
       .build();
     InputFileContext ctx = new InputFileContext(sensorContext, inputFile);
     visitor.scan(ctx, parser.parse(code, null));
     return visitor;
   }
+
+  protected String testFileName() {
+    return "file";
+  }
```

Comment:
You wouldn't need this if you decide on the language (see next comment). However, then you have to provide a language for the test file.